app/lib/actions/pulp/abstract_async_task.rb

```diff
@@ -79,7 +79,7 @@ def external_task
         output[:pulp_tasks]
       end
 
-      def cancel!
+      def cancel!(raise_error = true)
```

Comment:
That is correct.  The current dynflow polling mechanism only invokes three methods in an 'async task':

a) initiate the task
b) fetch the task details
c) check the task details to see if its done
d) repeat b & c until an error occurs or done? return true

We used to have the code to check for errors at b), but a dynflow change months ago caused dynflow to treat these as failed task polls rather than failure of the task.  Thus it was moved to done?.  I could see explicitly adding a check for errored? (very similar to done?) to be explicit about this, but that seems like side issue and may break backwards compatibility with older tasks depending on how it was implemented.   

If you prefer I think it may make more sense to split it up into a 'cancel!' and 'cancel' function, with the cancel function not calling done?.  Unless you have another thought